rapid7/metasploit-framework
# CHAOS rat xss to rce

CHAOS v5.0.8 is a free and open-source Remote Administration Tool that
allow generate binaries to control remote operating systems. The
webapp contains a remote command execution vulnerability which
can be triggered by an authenticated user when generating a new
executable. The webapp also contains an XSS vulnerability within
the view of a returned command being executed on an agent.

Execution can happen through one of three routes:

1. Provided credentials can be used to execute the RCE directly
2. A `JWT` token from an agent can be provided to emulate a compromised
host. If a logged in user attempts to execute a command on the host
the returned value contains an xss payload.
3. Similar to technique 2, an agent executable can be provided and the
`JWT` token can be extracted.

Verified against CHAOS `7d5b20ad7e58e5b525abdcb3a12514b88e87cef2` running
in a docker container.

## Verification

1. Install the application or run the docker image
1. Start msfconsole
1. Do: `use exploit/linux/http/chaos_rat_xss_to_rce`
1. Do: `set rhost [ip]`
1. Pick a method:
  1. `set username [username]`, `set password [password]`
  2. `set jwt [jwt token]`
  3. `set agent [path to agent]`
1. Do: `run`
1. You should get a shell. Interaction by a CHAOS admin may be required
